Sources/XcodeGenKit/SourceGenerator.swift

```diff
@@ -474,7 +474,7 @@ class SourceGenerator {
         var sourcePath = path
         switch type {
         case .folder:
-            let folderPath = Path(targetSource.path)
+            let folderPath = project.basePath + Path(targetSource.path)
```

Comment:
I had to change this line to fix a test that broke when I swapped out `byRemovingBase(path:)` with `relativePath(from:)`.

It seems like folder target sources were being created without the base directory as part of the path. In the tests, I saw that a folder "Sources/A" would passed to `getFileReference` as `Path("Sources/A")`, but a file "Sources/a.swift" would be passed as `Path("TestDirectory/Sources/a.swift")`.